### DISCUSSION TITLE
Error: Hydration failed because the initial UI does not match what was rendered on the server.

### DISCUSSION THREAD

**Discussion by kotAPI** [vercel/next.js#51868] [Help]
### Summary

```
Unhandled Runtime Error
Error: Hydration failed because the initial UI does not match what was rendered on the server.

Warning: Expected server HTML to contain a matching <section> in <div>.

See more info here: https://nextjs.org/docs/messages/react-hydration-error

Component Stack
section
BasicLayout
Adapter
Layout
QuickStartContextProvider

```

I am in the process of Upgrading my NextJS project to 13. I've fixed a lot of the hydration errors, I've been stuck with this one for a while and the errors shown by stack trace isn't really helpful. The error that you are seeing from the screenshot I think is being thrown by the underlying Antd library component and I do not have control over it. 

You can see there is no `section` tag used in the codebase, I've changed them all to divs because of the error
<img width="270" alt="image" src="https://github.com/vercel/next.js/assets/4013556/ee5e9ad5-9cba-45c3-9f77-0c1a8c1572cd">

<img width="957" alt="image" src="https://github.com/vercel/next.js/assets/4013556/798a4cc0-6198-4c5d-9288-997ab34d7fd4">

<img width="957" alt="image" src="https://github.com/vercel/next.js/assets/4013556/2a21dadd-60cd-477a-bbae-51d47f95dff7">
<img width="957" alt="image" src="https://github.com/vercel/next.js/assets/4013556/7d76a608-1e33-49f6-8046-33e5b63a90ea">

<img width="957" alt="image" src="https://github.com/vercel/next.js/assets/4013556/97b977c9-b661-496d-b09d-131772656ff6">

<img width="957" alt="image" src="https://github.com/vercel/next.js/assets/4013556/647669b3-b548-42ea-8168-20edd96f87bf">

### Additional information

```js
My project uses Antd UI library underneath - `4.15.2`
```

**Comment by icyJoseph** [comment]
Hi,

Do you use any component library? Looks like your BasicLayout does?

**Reply by icyJoseph** [reply]
I had missed the AntUI library information. 

What is the BasicLayout doing? It seems like it mutates the DOM in some strange way.

**Reply by kotAPI** [reply]
All nodes from `Layout` to `section` I think is being rendered by Antd library - Docs here- https://ant.design/components/layout. 

I too think Layout is the culprit, commenting it out seem to be silencing the hydration errors - but then again I cannot say for certain that it isn't any of the children components for my app that are causing it.

I had checked a lot of other answers before posting this here, is ant library known to cause such issues?

**Reply by icyJoseph** [reply]
Right, have you tried upgrading the antd library? perhaps they had some poor SSR support.

What version do you have? The link you shared is to the very last version.

**Reply by kotAPI** [reply]
I'm on 4.15.2 version of Antd, not sure of how good the support is for SSR 🤔

**Reply by kotAPI** [reply]
I haven't thought about SSR support from Antd, I think you're right... I haven't found any docs if latest antd version supports SSR 🤔 

some resources i found - 
https://stackoverflow.com/a/76520949/2818099
